app/src/main/java/com/lightricks/feedexercise/network/FeedApiService.kt

```diff
@@ -1,5 +1,22 @@
 package com.lightricks.feedexercise.network
 
-/**
- * todo: add the FeedApiService interface and the Retrofit and Moshi code here
- */
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+
+interface FeedApiService {
+    companion object {
+        const val BASE_URL: String = "https://assets.swishvideoapp.com/Android/demo/"
+        const val THUMBNAILS_URL: String = BASE_URL + "catalog/thumbnails/"
+        fun create(): FeedApiService {
+            val retrofit = Retrofit.Builder()
+                .addConverterFactory(MoshiConverterFactory.create())
+                .baseUrl(BASE_URL)
+                .build()
+            return retrofit.create(FeedApiService::class.java)
+        }
+    }
+
+    @GET("feed.json")
+    suspend fun getTemplatesMetadata(): TemplatesMetadata
+}
```

Comment:
eof

Reply:
end of file... what's to action required here?
add a newline?

Reply:
END OF FILE -> new line indeed.